src/app/dynamic-forms/fields/textbox/textbox.component.html

```diff
@@ -43,4 +43,8 @@
     <a *ngIf="field.value" [href]="field.value">Open</a>
     <label *ngIf="!field.value">-</label>
   </ng-container>
+
+  <ng-container *ngIf="showIFrame && isValidHttpUrl(field.value)">
+    <iframe class="iframe-control" [src]="urlSafe" [ngStyle] ="{'height.px' : iframeHeightStyle}"></iframe>
```

Comment:
@alvin-h Do we use any of the [security attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe) of iframes elsewhere in Cinchy apps and if so, should we do the same here? I'm not sure if someone using a url to a malicious app that could reach in and grab data from the parent page/form is a valid concern. We want to allow urls from any domain but ideally limit what the app in the source URL can do with the parent.

Reply:
It's a concern actually... if it's a cross domain website being rendered in the iframe, it's a lot safer, because it blocks access to the parent window, but I believe it could still redirect the parent's page to something malicious? Do we want to make it user configurable? We can either introduce CSP headers or do something with sandbox iframes, see this article: https://learn.microsoft.com/en-us/previous-versions/msdn10/hh563496(v=msdn.10)